### PR TITLE
US 02.04.01: Pull request

### LINKED ISSUE #23: US 02.04.01
As an organizer I want to upload an event poster to provide visual information to entrants.

- [x] Database needs to be connected for image uploads
- [x] Funtionality without database is complete

### PR DESCRIPTION
**Pull request by SuperCreativeUniqueUsername**
US 02.04.01: As an organizer I want to upload an event poster to provide visual information to entrants.

I added an icon when pressed allows the user to upload an image from the gallery. Todo: add database connection for image uploads

UML is updated:
image

Storyboard is updated with new add image icon: ( I couldn't figure out how to upload just the icon so I took a screenshot of the fragment and added that to the Storyboard diagram)
image

CRC doesn't need to be changed.

Testing wasn't updated since I couldn't figure out how to add images to the internal storage of a virtual device. When testing on my phone the uploading worked as expected.

Let me know if you can see anything I may have missed or needs to be changed 😌

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -53,4 +53,4 @@ dependencies {
     testImplementation("junit:junit:4.13.2")
     androidTestImplementation("androidx.test.ext:junit:1.1.5")
     androidTestImplementation("androidx.test.espresso:espresso-core:3.5.1")
-}
\ No newline at end of file
+}
```
Comment (katzhong) [inline app/build.gradle.kts:56]: just wondering, what was changed here?

File: app/src/main/java/com/example/pygmyhippo/organizer/PostEventFragment.java [hunk 2]
```diff
@@ -102,8 +111,35 @@ public void onClick(View v) {
...
+            public void onClick(View v) {
+                pickMedia.launch(new PickVisualMediaRequest.Builder()
+                        .setMediaType(ActivityResultContracts.PickVisualMedia.ImageOnly.INSTANCE)
+                        .build());
+            }
+        });
     }
 
+    /**
+     * Allowes the user to select an image for the event
```
Comment (katzhong) [inline app/src/main/java/com/example/pygmyhippo/organizer/PostEventFragment.java]: NIT: allowes --> allows

File: app/build.gradle.kts (hunk 1 above)
Comment (SuperCreativeUniqueUsername) [reply app/build.gradle.kts:56]: I had added the no lint checking flag, but when I removed it this changed for some reason.

File: app/src/main/res/layout/organiser_post_event.xml [hunk 3]
```diff
@@ -1,211 +1,217 @@
...
                 android:layout_height="60dp"
                 android:layout_marginTop="12dp"
                 android:checked="false"
-                android:textColor="@color/dark_purple"
-                android:textSize="18sp"
                 android:fontFamily="@font/roboto_mono_medium"
-                android:text="Enable Geolocation" />
+                android:text="Enable Geolocation"
+                android:textColor="@color/dark_purple"
+                android:textSize="18sp" />
```
Comment (katzhong) [inline app/src/main/res/layout/organiser_post_event.xml:197]: are these edits just to make them all in the same order?

File: app/src/main/java/com/example/pygmyhippo/organizer/PostEventFragment.java (hunk 2 above)
Comment (SuperCreativeUniqueUsername) [reply app/src/main/java/com/example/pygmyhippo/organizer/PostEventFragment.java]: changed

File: app/src/main/res/layout/organiser_post_event.xml (hunk 3 above)
Comment (SuperCreativeUniqueUsername) [reply app/src/main/res/layout/organiser_post_event.xml:197]: I dont even think i changed these, when I was playing around in the visual layout editor I think that's when it automatically reordered these.

Comment (katzhong) [reply app/src/main/res/layout/organiser_post_event.xml:197]: linting perhaps

**Review by katzhong — APPROVED** [review]

**Review by Kori-Kozicki — APPROVED** [review]
I think this looks pretty good. It worked fine for me and I had no issues. My only future question is can images be stored with the Uri class in firebase? Otherwise I think this is good to merge. (Also where did you update the storyboard? I'm having trouble finding any changes)

**Comment by SuperCreativeUniqueUsername** [comment]
> I think this looks pretty good. It worked fine for me and I had no issues. My only future question is can images be stored with the Uri class in firebase? Otherwise I think this is good to merge. (Also where did you update the storyboard? I'm having trouble finding any changes)

Were probably going to need a different method when we start using the firebase storage for images instead of URI, I would need to update the image uploading after we make that change. We would probably have to store a reference (basically the file path to the image in the firebase storage) in the event class, when we want to display this image we'd probably make an image handler class that could pull that image for us to display. For the storyboarding I only updated the logo the empty event image.
image

**Pull request merged**